pallets/issue/src/lib.rs

```diff
@@ -674,4 +704,46 @@ impl<T: Config> Pallet<T> {
 	fn issue_minimum_transfer_amount(currency_id: CurrencyId<T>) -> Amount<T> {
 		Amount::new(IssueMinimumTransferAmount::<T>::get(), currency_id)
 	}
+
+	fn increase_interval_volume(issue_amount: Amount<T>) -> Result<(), DispatchError> {
+		Ok(if let Some(limit_volume) = LimitVolumeAmount::<T>::get() {
+			let issue_volume = Self::convert_into_limit_currency_id_amount(issue_amount)?;
+			let current_volume = CurrentVolumeAmount::<T>::get();
+			let new_volume = current_volume
+				.checked_add(&issue_volume.amount())
+				.ok_or(ArithmeticError::Overflow)?;
+			CurrentVolumeAmount::<T>::put(new_volume);
+		})
+	}
+
+	fn convert_into_limit_currency_id_amount(
+		issue_amount: Amount<T>,
+	) -> Result<Amount<T>, DispatchError> {
+		let issue_volume =
+			oracle::Pallet::<T>::convert(&issue_amount, LimitVolumeCurrencyId::<T>::get())
+				.map_err(|_| DispatchError::Other("Missing Exchange Rate"))?;
+		Ok(issue_volume)
+	}
+
+	fn check_volume(amount_requested: Amount<T>) -> Result<(), DispatchError> {
+		let limit_volume: Option<BalanceOf<T>> = LimitVolumeAmount::<T>::get();
+		Ok(if let Some(limit_volume) = limit_volume {
+			let current_block: T::BlockNumber = <frame_system::Pallet<T>>::block_number();
+			let interval_length: T::BlockNumber = IntervalLength::<T>::get();
+
+			let current_index = current_block.checked_div(&interval_length);
+			let mut current_volume = BalanceOf::<T>::default();
+			if current_index > Some(CurrentInterval::<T>::get()) {
+				CurrentInterval::<T>::put(current_index.unwrap_or_default());
+				CurrentVolumeAmount::<T>::put(current_volume);
+			} else {
+				current_volume = CurrentVolumeAmount::<T>::get();
+			}
+			let new_issue_request = Self::convert_into_limit_currency_id_amount(amount_requested)?;
+			ensure!(
+				new_issue_request.amount().saturating_add(current_volume) > limit_volume,
+				Error::<T>::ExceedLimitVolumeForIssueRequest
+			);
```

Comment:
This statement needs to be flipped actually. Like this, it would mean that the error is not thrown when the amounts are above the limit_volume which is exactly the opposite of what we want.

Reply:
So it should be 
```
				new_issue_request.amount().saturating_add(current_volume) <= limit_volume,
```
instead